tests/director/tests/formation_api_test.go

```diff
@@ -564,6 +564,122 @@ func TestTenantFormationFlow(t *testing.T) {
+		TargetOperation:     graphql.TargetOperationAssignFormation,
+		Operator:            "IsNotAssignedToAnyFormationOfType",
+		ResourceType:        graphql.ResourceTypeTenant,
+		ResourceSubtype:     "subaccount",
+		OperatorScope:       graphql.OperatorScopeTenant,
+		InputTemplate:       "{\\\"formation_template_id\\\": \\\"{{.FormationTemplateID}}\\\",\\\"resource_type\\\": \\\"{{.ResourceType}}\\\",\\\"resource_subtype\\\": \\\"{{.ResourceSubtype}}\\\",\\\"resource_id\\\": \\\"{{.ResourceID}}\\\",\\\"tenant\\\": \\\"{{.TenantID}}\\\"}",
+		ConstraintScope:     graphql.ConstraintScopeFormationType,
+		FormationTemplateID: formationTemplateOutput.ID,
+	}
+	constraint := fixtures.CreateFormationConstraint(t, ctx, certSecuredGraphQLClient, in)
+	defer fixtures.CleanupFormationConstraint(t, ctx, certSecuredGraphQLClient, constraint.ID)
+	require.NotEmpty(t, constraint.ID)
+
+	t.Logf("Should create formation: %s", firstFormation)
+
+	formationInput := fixtures.FixFormationInput(firstFormation, str.Ptr(formationTemplateName))
+	formationInputGQL, err := testctx.Tc.Graphqlizer.FormationInputToGQL(formationInput)
+	require.NoError(t, err)
+
+	var formation graphql.Formation
+	createReq := fixtures.FixCreateFormationWithTemplateRequest(formationInputGQL)
+	defer fixtures.DeleteFormation(t, ctx, certSecuredGraphQLClient, firstFormation)
+	err = testctx.Tc.RunOperation(ctx, certSecuredGraphQLClient, createReq, &formation)
+	require.NoError(t, err)
+	require.Equal(t, firstFormation, formation.Name)
+
+	t.Logf("Should create formation: %s", secondFormation)
+
+	secondFormationInput := fixtures.FixFormationInput(secondFormation, str.Ptr(formationTemplateName))
+	secondFormationInputGQL, err := testctx.Tc.Graphqlizer.FormationInputToGQL(secondFormationInput)
+	require.NoError(t, err)
+
+	var unusedFormation graphql.Formation
+	createUnusedReq := fixtures.FixCreateFormationWithTemplateRequest(secondFormationInputGQL)
+	defer fixtures.DeleteFormation(t, ctx, certSecuredGraphQLClient, secondFormation)
+	err = testctx.Tc.RunOperation(ctx, certSecuredGraphQLClient, createUnusedReq, &unusedFormation)
+	require.NoError(t, err)
+	require.Equal(t, secondFormation, unusedFormation.Name)
+
+	t.Logf("Assign tenant %s to formation %s", subaccountID, firstFormation)
+	assignReq := fixtures.FixAssignFormationRequest(subaccountID, string(graphql.FormationObjectTypeTenant), firstFormation)
+	var assignFormation graphql.Formation
+	err = testctx.Tc.RunOperation(ctx, certSecuredGraphQLClient, assignReq, &assignFormation)
+	require.NoError(t, err)
+	require.Equal(t, firstFormation, assignFormation.Name)
+
+	saveExampleInCustomDir(t, assignReq.Query(), assignFormationCategory, "assign tenant to formation")
+
+	t.Log("Should match expected ASA")
+	asaPage := fixtures.ListAutomaticScenarioAssignmentsWithinTenant(t, ctx, certSecuredGraphQLClient, tenantId)
+	require.Equal(t, 1, len(asaPage.Data))
+
+	assignment := graphql.AutomaticScenarioAssignmentSetInput{
```

Comment:
Add unassign defer here as well in case we have a bug and it does not fail to have robust clean ups.